src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,48 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import mate.academy.exception.NoUserInCacheException;
+import mate.academy.exception.RequestProcessingException;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(
+                () -> findUserInCache(userId), executor).exceptionally(ex -> {
+                    try {
+                        UserData userData = emulateDataBaseAccess(userId);
+                        cacheFetchedData(userId, userData);
+                        return userData;
```

Comment:
most of this logic can fit into an elegant [`cache.computeIfAbsent()`](https://www.baeldung.com/java-map-computeifabsent) call